stackit/internal/services/iaas/publicipassociate/resource.go

```diff
@@ -0,0 +1,360 @@
+func (r *publicIpAssociateResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) { // nolint:gocritic // function signature required by Terraform
+	var model Model
+	diags := req.State.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	projectId := model.ProjectId.ValueString()
+	publicIpId := model.PublicIpId.ValueString()
+	networkInterfaceId := model.NetworkInterfaceId.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "public_ip_id", publicIpId)
+	ctx = tflog.SetField(ctx, "network_interface_id", networkInterfaceId)
+
+	publicIpResp, err := r.client.GetPublicIP(ctx, projectId, publicIpId).Execute()
+	if err != nil {
+		oapiErr, ok := err.(*oapierror.GenericOpenAPIError) //nolint:errorlint //complaining that error.As should be used to catch wrapped errors, but this error should not be wrapped
+		if ok && oapiErr.StatusCode == http.StatusNotFound {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error reading public IP associate ", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+
+	// Map response body to schema
+	err = mapFields(publicIpResp, &model)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error reading public IP associate", fmt.Sprintf("Processing API payload: %v", err))
+		return
+	}
+	// Set refreshed state
+	diags = resp.State.Set(ctx, model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	tflog.Info(ctx, "public IP associate read")
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *publicIpAssociateResource) Update(_ context.Context, _ resource.UpdateRequest, _ *resource.UpdateResponse) { // nolint:gocritic // function signature required by Terraform
+	// Update is not supported, all fields require replace
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *publicIpAssociateResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve values from state
+	var model Model
+	diags := req.State.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectId := model.ProjectId.ValueString()
+	publicIpId := model.PublicIpId.ValueString()
+	networkInterfaceId := model.NetworkInterfaceId.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "public_ip_id", publicIpId)
+	ctx = tflog.SetField(ctx, "network_interface_id", networkInterfaceId)
+
+	// Delete existing publicIp
+	err := r.client.DeletePublicIP(ctx, projectId, publicIpId).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting public IP associate", fmt.Sprintf("Calling API: %v", err))
```

Comment:
```suggestion
		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting public IP association", fmt.Sprintf("Calling API: %v", err))
```

Reply:
Actually, this really deletes the public IP, right? We should with the IaaS team that this is intended

Reply:
as agreed with IaaS team, we will call the update api and give network_interface nil